runtime/compiler/control/CompilationThread.cpp

```diff
@@ -6035,6 +6070,44 @@ void *TR::CompilationInfo::compileOnSeparateThread(J9VMThread * vmThread, TR::Il
 
       entry->releaseSlotMonitor(vmThread);
 
+#if DEV_DBG_ENABLE_CHANGE
+      // TODO: DO we need to grab these mutex?
+      // entry->acquireSlotMonitor(vmThread)
+      // Should this be before or after acquireVMAccess(vmThread)?
+      //
+      // If a remote sync compilation has been changed to a local sync compilation,
+      // queue a new entry for this method as a remote async compilation
+      acquireCompMonitor(vmThread);
+      if ((startPC && startPC != oldStartPC) &&
+           entry->_changedToLocalSyncComp)
+         {
+         TR::IlGeneratorMethodDetails methodDetails(method);
+         CompilationPriority compPriority = CP_ASYNC_NORMAL;
+         bool queuedForRemote = false;
+         bool isAsync = true;
+         TR_OptimizationPlan *plan = TR_OptimizationPlan::alloc(entry->_optimizationPlan->getOptLevel());
```

Comment:
Don't we need to take the optLevel from optimizationPlan->_origOptLevel?

Reply:
The entry's level is restored in `postCompilationTasks()`:
https://github.com/eclipse/openj9/compare/jitaas..4f62ac3ab1e93a140183dd71f62ef13d1f0fa61f#diff-08779930c70bee9d159bc2c125154c30R7120

We could also directly use `entry->_origOptLevel` here to create the new optimization plan.